cmd.go

```diff
@@ -343,11 +343,23 @@ func handleCommandError(c Command, ctx *Context, err error, f *gnuflag.FlagSet)
 	case nil:
 		return 0, false
 	case gnuflag.ErrHelp:
-		ctx.Stdout.Write(c.Info().Help(f))
+		_, err := ctx.Stdout.Write(c.Info().Help(f))
+		if err != nil {
+			logger.Errorf("problem trying to write to stdout: %v", err)
+		}
 		return 0, true
 	case ErrSilent:
 		return 2, true
 	default:
+		name := parseFlag("format", f.Args())
```

Comment:
The "problem" is that cmd.Main does not have access to the state of it's subcommands as we rely on the interface being passed in. Therefore we only have access to the f.Args(), which in this case get's parsed.

Another alternative Option would be to add the below function to the gnuflag dependency.